crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -386,6 +386,8 @@ impl PartialEq for EntityRef<'_> {
 
 impl Eq for EntityRef<'_> {}
 
+// We intentionally choose to have EntityRef compare like Entity.
```

Comment:
I suppose the danger here is that entities from different `world`s that happen to have the same ID will compare equal.  That seems unlikely to be a problem in practice, although it might be worth documenting somewhere.

Reply:
Yeah, it is a bit unfortunate that we can't have both, but it seems to me that being able to compare them like `Entity` is a lot more valuable. I think I'll add some documentation to the types themselves.